mathcomp/ssreflect/plugin/trunk/ssreflect.ml4

```diff
@@ -2093,10 +2090,15 @@ let abs_wgen keep_let ist f gen (gl,args,c) =
   match gen with
   | _, Some ((x, mode), None) when mode = "@" || (mode = " " && keep_let) ->
      let x = hoi_id x in
-     let _, bo, ty = Named.Declaration.to_tuple (pf_get_hyp gl x) in
-     gl,
-     (if bo <> None then args else mkVar x :: args),
-     mkProd_or_LetIn (mk_reldecl (Name (f x)) bo ty) (subst_var x c)
+     (match pf_get_hyp gl x with
+     | LocalAssum (_,ty) ->
+         gl,
+         mkVar x :: args,
+         mkProd_or_LetIn (RelDecl.LocalAssum (Name (f x),ty)) (subst_var x c)
+     | LocalDef (_,b,ty) ->
+         gl,
+         args,
+         mkProd_or_LetIn (RelDecl.LocalDef (Name (f x),b,ty)) (subst_var x c))
   | _, Some ((x, _), None) ->
```

Comment:
Sorry but I don't like such duplication of code, this is why I did not merge this part of your previous pull request (and defined the mk_reldecl function).